JBIDE-22265 Application wizard: Allow users to reset values of all template parameters

scabanovich opened the pull request:
1. Button 'Reset All' is added.
2. Modified values are rendered in italics.
3. Buttons 'Reset' and 'Reset All' are enabled only when they can reset something.

scabanovich commented:
@fbricon please take a look at the pull request.

fbricon commented on plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/utils/TableViewerBuilder.java:
```diff
@@ -50,13 +51,14 @@ public TableViewerBuilder(TableViewer viewer, Composite tableContainer) {
+		FontData fd = normalFont.getFontData()[0];
 		fd.setStyle(SWT.ITALIC);
-		italic = new Font(null, fd);
+		italicFont = new Font(null, fd);
```
I use JFaceResources.getFontRegistry().getItalic(JFaceResources.DEFAULT_FONT)

According to http://stackoverflow.com/questions/1830058/styling-a-swt-label-to-be-italic#comment42024249_18272783 you don't need to dispose it yourself

scabanovich replied:
Can we be sure that Table.getFont() returns the same font as JFaceResources.DEFAULT_FONT?

fbricon replied:
I have no idea. But doesn't it make sense the default font for the table would actually be JFaceResources.DEFAULT_FONT?

I'm not fighting to impose using that approach anyway. Just letting you know it's a possibility. If your approach makes more sense, then keep at it.

scabanovich replied:
There is also JFaceResources.DIALOG_FONT that may be more relevant as we have a dialog in this case.
I checked that in my environment JFaceResources.getDefaultFont(), JFaceResources.getDialogFont() and Table.getFont() return fonts with the same name, height and style, but that can be different in other environments.

And, for an experiment I made a call JFaceResources.getDialogFont().dispose(). Wow, it broke all dialogs Eclipse!

jeffmaury commented:
Wouldn't it be more logical when a parameter is updated to fire an change event rather than refreshing the whole table ?

adietish commented:
@jeffmaury if I get your comment right then you're talking of the current impl where the table is refreshed upon "reset all": https://github.com/jbosstools/jbosstools-openshift/pull/1134/files#diff-697b5236327200e5b04ac283fa2f7082R430 which calls table.refresh() here https://github.com/jbosstools/jbosstools-openshift/pull/1134/files#diff-697b5236327200e5b04ac283fa2f7082R447
I basically agree with your comment as in MVC, but this requires the table to be implemented with binding or a classical listener pattern. I dont see this having been done: https://github.com/jbosstools/jbosstools-openshift/pull/1134/files#diff-697b5236327200e5b04ac283fa2f7082R209 So this means that the impl would have to be changed to achieve this. Maybe file an enhancement in jira?

jeffmaury commented:
What I mean is that when a single parameter is modified, it does not seem to me that the whole table is refreshed. So basically Refresh all is no more than a set of parameter modification

scabanovich commented:
Ok, I removed that refresh and added a listener that refreshes only the changed item in the table.
The table is already bound to list property "parameters" - but table responds only to changes in the list, not to changes inside an element when instance remains the same. Hence, yes, the new listener is added in a rather classical way.
